src/libraries/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -1640,6 +1690,71 @@ private static void TestExportStore(X509ContentType ct)
             }
         }
 
+        private static void AssertPemExport(X509Certificate2Collection collection, string expectedContents)
+        {
+            Span<char> buffer = new char[expectedContents.Length];
+            int written;
+
+            // If we expect something to get written, try writing to a buffer that is too small by one
+            // and make sure it fails.
+            if (!buffer.IsEmpty)
+            {
+                Assert.False(collection.TryExportCertificatePems(buffer.Slice(1), out written), nameof(collection.TryExportCertificatePems));
+                Assert.Equal(0, written);
+            }
+
+            Assert.True(collection.TryExportCertificatePems(buffer, out written), nameof(collection.TryExportCertificatePems));
+            Assert.Equal(written, buffer.Length);
+            Assert.Equal(expectedContents, new string(buffer));
```

Comment:
Since this test is looking for an exact export it's already covering

* No gunk in the header
* No extra newline at the end
* No gunk in the footer
* No gunk between elements.

It's a very expressive test :smile: